pkg/operator/ceph/reporting/reporting.go

```diff
@@ -33,25 +34,75 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+// get the kind through the object API, but if that is empty, make a best guess via golang reflection
+func objKindOrBestGuess(obj client.Object) string {
+	if obj == nil {
+		return "<nil>"
+	}
+
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	if kind != "" {
+		return kind
+	}
+
+	t := reflect.TypeOf(obj)
+	if t.Kind() == reflect.Ptr {
+		return t.Elem().Name()
+	}
+	return t.Name()
+}
+
 // ReportReconcileResult will report the result of an object's reconcile in 2 ways:
 // 1. to the given logger
 // 2. as an event on the object (via the given event recorder)
-// The results of the object's reconcile should include the object, the reconcile response, and the
-// error returned by the reconcile.
+//
+// The results of the object's reconcile should include the object (NEVER NIL),
+// the reconcile response, and the error returned by the reconcile.
+//
+// The reconcile request is used to reference a given object that doesn't have a name.
+//
 // The function is designed to return the appropriate values needed for the controller-runtime
 // framework's Reconcile() method.
-func ReportReconcileResult(logger *capnslog.PackageLogger, recorder record.EventRecorder,
-	obj client.Object, reconcileResponse reconcile.Result, err error,
+func ReportReconcileResult(
+	logger *capnslog.PackageLogger,
+	recorder record.EventRecorder,
+	reconcileRequest reconcile.Request,
+	obj client.Object,
+	reconcileResponse reconcile.Result,
+	err error,
 ) (reconcile.Result, error) {
-	kind := obj.GetObjectKind().GroupVersionKind().Kind
-	nsName := fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName())
+	kind := objKindOrBestGuess(obj)
+	nsName := reconcileRequest.NamespacedName.String()
+
+	var objCopy client.Object
+
+	// if object is nil, it is an error, but we can't prevent caller from doing the wrong thing here
+	if obj == nil {
+		// TODO: does this still blow up if obj is nil?
```

Comment:
Do you need to return if `obj == nil`? Not sure anything will blow up, but it will result on an empty object. What does that look like?